plugins/inputs/aerospike/aerospike.go

```diff
@@ -151,6 +153,58 @@ func (a *Aerospike) gatherServer(hostport string, acc telegraf.Accumulator) erro
 			}
 			acc.AddFields("aerospike_namespace", nFields, nTags, time.Now())
 		}
+
+		latencyInfo, err := as.RequestNodeInfo(n, "latency:")
+		if err != nil {
+			return err
+		}
+		latency, exists := latencyInfo["latency:"]
+		if !exists {
+			return errors.New("Aerospike Input Plugin : No key `latency:` in response of as.RequestNodeInfo(n, \"latency:\") from Aerospike Server")
+		}
+		points := strings.Split(latency, ";")
+		i := 0
+		for i < len(points) {
+			point := points[i]
+			if strings.Compare(point, "error-no-data-yet-or-back-too-small") == 0 {
+				i++
+			} else {
+				indexOfColon := strings.Index(point, ":")
+				if indexOfColon == -1 {
+					i++
+					continue
+				}
+				histogram := point[0:indexOfColon]
+				operation, namespace := splitNamespaceAndOperation(histogram)
+				latencyKey := point[indexOfColon+1:]
+				latencyValue := points[i+1]
+				keys := strings.Split(latencyKey, ",")
+				values := strings.Split(latencyValue, ",")
+				ltags := map[string]string{
+					"aerospike_host": hostport,
+					"node_name":      n.GetName(),
+					"namespace":      namespace,
+				}
+				lfields := make(map[string]interface{})
+				var duration, totalOperations float64
+				duration = getDurationOfTransaction(keys[0], values[0])
+				operationsPerSecond := parseValue(values[1])
+				if operationsPerSecond.(float64) > 0 && len(keys) == len(values) {
```

Comment:
It seems like we are pretty sure this is a float?  If so, we shouldn't use parserValue and do the conversion here.

Reply:
Yes, value float type all the time. let me remove the parseValue and conversion.